### DISCUSSION TITLE
singularity brainvisa multiusers

### DISCUSSION THREAD

**Discussion by lemaitreh** [brainvisa/brainvisa.github.io#39] [Q&A] [answered]
Hi guys ,

I followed the instruction on https://brainvisa.info/web/download.html
and installed brainvisa 5.0.1 using singularity 3.7.3 on centos7.

I works well with root but when i use another account:

lemaitre@pipeau:~$ brainvisa
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-lemaitre'
Loading toolbox brainrat
Loading toolbox cortical_surface
Loading toolbox data management
Loading toolbox datamind
Loading toolbox freesurfer
Loading toolbox morphologist
Loading toolbox primatologist
Loading toolbox spm
Loading toolbox structural_analysis
Loading toolbox tools
Loading toolbox viewers
Loading toolbox My processes
Fail to get or set current runs information:
[Errno 13] Permission denied: '/casa/home/.brainvisa/current_runs.minf'
Traceback (most recent call last):
  File "<string>", line 32, in <module>
  File "/casa/install/brainvisa/neuro.py", line 245, in <module>
    main()
  File "/casa/install/brainvisa/neuro.py", line 156, in main
    axon.initializeProcesses()
  File "/casa/install/python/brainvisa/axon/processes.py", line 132, in initializeProcesses
    dontrecordruns=dontrecordruns)
  File "/casa/install/python/brainvisa/configuration/neuroConfig.py", line 852, in __init__
    os.remove(self.file)
OSError: [Errno 13] Permission denied: '/casa/home/.brainvisa/current_runs.minf'
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/casa/install/python/brainvisa/axon/processes.py", line 81, in cleanup
    neuroHierarchy.databases.currentThreadCleanup()
AttributeError: 'NoneType' object has no attribute 'currentThreadCleanup'
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/casa/install/python/brainvisa/axon/processes.py", line 81, in cleanup
    neuroHierarchy.databases.currentThreadCleanup()
AttributeError: 'NoneType' object has no attribute 'currentThreadCleanup'

So, I changed the permission to be writable for everybody but i'm wondering if it is the correct setup for a multiuser system.

Hervé

**Comment by denisri** [comment, marked as answer]
If I understand it's a "shared" install for multiple users, right ? The permission issues occur because the home directory in /casa/host/home is used by multiple users. The solution is... undocumented yet (I think), and is:
Remove the "home" directory from the shared install directory (as admin). In this situation casa-distro will create a new "container home" directory in users home directories, in ~/.local/share/casa-distro/ so that each user will get a personal directory for himself, with appropriate permissions.

**Reply by lemaitreh** [reply]
Yes!!! That's it!!! Thanks
